.github/workflows/push.yml

```
@@ -11,6 +11,27 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  upload-test-coverage:
+    name: Upload Test Coverage
```

Is this the Unit tests conventional job? It's missing permissions...

```suggestion
  test-unit:
    name: Unit tests
    permissions:
      contents: read
```